_posts/2023-05-25-Announcing-Data-Prepper-2.3.0.md

```diff
@@ -0,0 +1,157 @@
+---
+
+Data Prepper 2.3.0 is now available for [download](https://opensearch.org/downloads.html#data-prepper)!
+This release introduces a number of changes that improve Data Prepper’s ability to process complex expressions with functions, arithmetic operations, and string operations.
+
+## Enhancements to Data Prepper expressions
+
+Data Prepper 2.3 supports using functions in expressions. A list of supported functions is available in the [Expression syntax](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/) documentation.
+
+Data Prepper 2.3 supports three types of expressions; [conditional](#conditional-expressions), [arithmetic](#arithmetic-expressions), and [string](#string-expressions).
+
+### Conditional expressions
+
+ Conditional expressions evaluate to a result of Boolean type, and the expressions can now have functions in them. For example, `length(/message) > 20` would evaluate to true if the length of the message field in the event is greater than 20. Otherwise it evaluates to false.
+
+### Arithmetic expressions
+
+Arithmetic expressions evaluate to a result of integer or float type. The expressions can have simple arithmetic operators like `+,-,*,/`, with functions, JSON pointers, or literals as operands. For example, the following expression adds the length of `message` field in the event with the value of `integerField` in the event metadata and subtracts 1 from it.
+
+```
+length(/message) + getMetadata("integerField") - 1
+```
+
+### String expressions
+
+String expressions evaluate to a result of string type. The string concatenation operator is supported, and it uses functions, JSON pointers, or literals as operands. For example, the following expression concatenates the `message1` field of type string in the event with the `message2` field in the event and appends "suffix" to it:
+
+```
+/message1 + /message2 + "suffix"
+```
+
+## Event tagging
+
+Data Prepper 2.3 supports tagging events while using the `grok` processor. Events can be tagged optionally be tagged using the following configuration:
+
+```
+processor:
+   - grok:
+        match:
+          message: <pattern>
+        tags_on_match_failure: ["grok_match_fail", "another_tag"]
+```
+
+The presence of tags can be checked in conditional expressions in different processors or routing using the `hasTags()` function to perform conditional ingestion. For example, a conditional expression checking for the presence of tag `grok_match_fail` would be `hasTags("grok_match_fail")`.
+
+## Enhancements to `add_entries` processor
+
+The `add_entries` processor has been enhanced to support values with expressions and to set event metadata keys.
+
+### Expression-based value
+
+The `add_entries` processor has been enhanced to support adding values based on an expression where the return type can be Boolean, integer or float, or string. The following example shows an `add_entries` processor configuration with a `value_expression` option:
```

Comment:
Consider linking add entries processor to here: https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/processors/add-entries/